src/Support/InterfaceType.php

```diff
@@ -39,6 +54,11 @@ public function getAttributes(): array
             $attributes['resolveType'] = $resolver;
         }
 
+        $resolver = $this->getTypesResolver();
+        if ($resolver) {
+            $attributes['types'] = $resolver;
+        }
+
```

Comment:
We need to add a method in each `Interface`: `types` returning an array of `Type`s. This way, we know what `Type`s to expect while executing our relation's query.